src/algebraic_geometry/gluing.lean

```diff
@@ -0,0 +1,227 @@
+structure glue_data extends category_theory.glue_data Scheme :=
+(f_open : ∀ i j, is_open_immersion (f i j))
+
+attribute [instance] glue_data.f_open
+
+namespace glue_data
+
+variables (D : glue_data)
+
+include D
+
+local notation `𝖣` := D.to_glue_data
+
+/-- The glue data of locally ringed spaces spaces associated to a family of glue data of schemes. -/
+abbreviation to_LocallyRingedSpace_glue_data : LocallyRingedSpace.glue_data :=
+{ f_open := D.f_open,
+  to_glue_data := 𝖣 .map_glue_data forget_to_LocallyRingedSpace }
+
+/-- (Implementation). The glued scheme of a glue data.
+This should not be used outside this file. Use `Scheme.glue_data.glued` instead. -/
+def glued_Scheme : Scheme :=
+begin
+  apply LocallyRingedSpace.is_open_immersion.Scheme
+    D.to_LocallyRingedSpace_glue_data.to_glue_data.glued,
+  intro x,
+  obtain ⟨i, y, rfl⟩ := D.to_LocallyRingedSpace_glue_data.ι_jointly_surjective x,
+  refine ⟨_, _ ≫ D.to_LocallyRingedSpace_glue_data.to_glue_data.ι i, _⟩,
+  swap, exact (D.U i).affine_cover.map y,
+  split,
+  { dsimp,
+    rw [coe_comp, set.range_comp],
+    refine set.mem_image_of_mem _ _,
+    exact (D.U i).affine_cover.covers y },
+  { apply_instance },
+end
+
+instance : creates_colimit 𝖣 .diagram.multispan forget_to_LocallyRingedSpace :=
+creates_colimit_of_fully_faithful_of_iso D.glued_Scheme
+  (has_colimit.iso_of_nat_iso (𝖣 .diagram_iso forget_to_LocallyRingedSpace).symm)
+
+instance : preserves_colimit 𝖣 .diagram.multispan forget_to_Top :=
+begin
+  delta forget_to_Top LocallyRingedSpace.forget_to_Top,
+  apply_instance,
+end
+
+instance : has_multicoequalizer 𝖣 .diagram :=
+has_colimit_of_created _ forget_to_LocallyRingedSpace
+
+/-- The glued scheme of a glued space. -/
+abbreviation glued : Scheme := 𝖣 .glued
+
+/-- The immersion from `D.U i` into the glued space. -/
+abbreviation ι (i : D.J) : D.U i ⟶ D.glued := 𝖣 .ι i
+
+/-- The gluing as sheafed spaces is isomorphic to the gluing as presheafed spaces. -/
+abbreviation iso_LocallyRingedSpace :
+  D.glued.to_LocallyRingedSpace ≅ D.to_LocallyRingedSpace_glue_data.to_glue_data.glued :=
+𝖣 .glued_iso forget_to_LocallyRingedSpace
+
+lemma ι_iso_LocallyRingedSpace_inv (i : D.J) :
+  D.to_LocallyRingedSpace_glue_data.to_glue_data.ι i ≫ D.iso_LocallyRingedSpace.inv = 𝖣 .ι i :=
+𝖣 .ι_glued_iso_inv forget_to_LocallyRingedSpace i
+
+instance ι_is_open_immersion (i : D.J) :
+  is_open_immersion (𝖣 .ι i) :=
+by { rw ← D.ι_iso_LocallyRingedSpace_inv, apply_instance }
+
+lemma ι_jointly_surjective (x : 𝖣 .glued.carrier) :
+  ∃ (i : D.J) (y : (D.U i).carrier), (D.ι i).1.base y = x :=
+𝖣 .ι_jointly_surjective (forget_to_Top ⋙ forget Top) x
+
+@[simp, reassoc]
+lemma glue_condition (i j : D.J) :
+  D.t i j ≫ D.f j i ≫ D.ι j = D.f i j ≫ D.ι i :=
+𝖣 .glue_condition i j
+
+/-- The pullback cone spanned by `V i j ⟶ U i` and `V i j ⟶ U j`.
+This is a pullback diagram (`V_pullback_cone_is_limit`). -/
+def V_pullback_cone (i j : D.J) : pullback_cone (D.ι i) (D.ι j) :=
```

Comment:
Is this part of the public API? If so, I don't really like the explicit mention of `V` in the name. Maybe just call it `pullback_cone`?

Reply:
It is a part of the API. 
But given a `D : glue_data`, I don't think I would have guessed what `D.pullback_cone` means.